### PR TITLE
[New] Add material-kit@1.1.0 w/ git auto-update

### LINKED ISSUE #11358: [Request] Add material-kit
**Library name:** material-kit
**Git repository url:** 
**npm package name or url** (if there is one): 
**License (List them all if it's multiple):** 
**Official homepage:** 
**Wanna say something? Leave message here:** 

=====================

Notes from cdnjs maintainer:
Please read the [README.md](https://github.com/cdnjs/cdnjs#cdnjs-library-repository) and [CONTRIBUTING.md](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md) document first.

We encourage you to add a library via sending pull request,
it'll be faster than just opening a request issue,
since there are tons of issues, please wait with patience,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
close #11358 
Pull request for issue: #11358
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/creativetimofficial/material-kit
 * Official website (optional, not the repository): https://www.creative-tim.com/product/material-kit
 * NPM package url (optional): 
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 61
   - Count of stars: 2199
   - Count of forks: 207
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
@dakshshah96 @cdnjs/intern3 
Please help me review this PR, Thank you.

File: ajax/libs/material-kit/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "keywords": [
+    "material",
+    "bootstrap",
+    "design",
+    "google",
+    "twitter"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https+git://github.com/creativetimofficial/material-kit.git"
```
Comment (extend1994) [inline ajax/libs/material-kit/package.json]: Should this be `git+https://github.com/creativetimofficial/material-kit.git`?

File: ajax/libs/material-kit/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/creativetimofficial/material-kit.git",
+    "fileMap": [
+      {
+        "basePath": "assets",
+        "files": [
+          "+(js|css)/material-kit*"
```
Comment (extend1994) [inline ajax/libs/material-kit/package.json]: We also need img/arrow-left.png and img/arrow-right.png. Refer https://github.com/creativetimofficial/material-kit/blob/master/assets/css/material-kit.css#L2775 and https://github.com/creativetimofficial/material-kit/blob/master/assets/css/material-kit.css#L2778.

**Comment by sufuf3** [comment]
@extend1994 I modified the package.json and add the files
Please help me review this PR, Thank you.

File: ajax/libs/material-kit/package.json (hunk 1 above)
Comment (mercyg) [inline ajax/libs/material-kit/package.json]: @extend1994 @sufuf3  Why do we need the `git+https://github.com/creativetimofficial/material-kit.git?`. The URL  shouldn't it be just "https://github.com/creativetimofficial/material-kit.git"? In the other libraries that were added was like this..?

**Comment by sufuf3** [comment]
@alexandru-paduraru Thanks that pay attention to this PR.
I'd like let CDNJS host your library, it's helpful to web front-end developer.
I just waiting for reviewing. If the reviewers have some questions about this library. This PR will need your help.

Thank you.

File: ajax/libs/material-kit/package.json (hunk 1 above)
Comment (sufuf3) [inline ajax/libs/material-kit/package.json]: `gt clone git://github.com/creativetimofficial/material-kit.git` can also download the library. :)

**Comment by alexandru-paduraru** [comment]
@sufuf3 that sounds really cool! Thank you. So let's do this! :D

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

File: ajax/libs/material-kit/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/creativetimofficial/material-kit.git",
+    "fileMap": [
+      {
+        "basePath": "assets",
+        "files": [
+          "+(js|css|img)/+(material-kit*|arrow*)"
```
Comment (PeterDaveHello) [inline ajax/libs/material-kit/package.json]: Should not use `arrow` as pattern for `img` but `png` or `cur`, actually we can grab all `img` here which would be even better and easier, simpler.

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3
I updated the pattern and I didn't include all img files because I don't think we need to add JPG files, which is used by examples page https://github.com/creativetimofficial/material-kit/blob/master/examples/profile-page.html#L71

Please help me review this PR, Thank you.

**Comment by alexandru-paduraru** [comment]
I think we can add it without images, they are also changed by the customers with their own images and we can support the hosting for them. No worries.

If there is something else that we need to do please let me know.

Best,
Alex

**Comment by PeterDaveHello** [comment]
@alexandru-paduraru thanks a lot, we'll take a look!

**Comment by sufuf3** [comment]
@alexandru-paduraru Thank you!
May I ask that is it OK to exclude `arrow*` image files?
Because I saw the files include in CSS file https://github.com/creativetimofficial/material-kit/blob/master/assets/css/material-kit.css#L2775
Thank you.

**Comment by alexandru-paduraru** [comment]
@sufuf3 yes, you can exclude them. We will prepare an update that will remove them from CSS too and move them to HTML or replace them with font icons. 

Best,
Alex

**Comment by sufuf3** [comment]
@alexandru-paduraru Thank you.
I'll update this PR. :)

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3
I updated the pattern and removed the `img` folder.
Please help me review this PR, Thank you.

**Comment by alexandru-paduraru** [comment]
@extend1994 material.min.js is the library that comes from Bootstrap Material Design: https://cdnjs.com/libraries/bootstrap-material-design is it ok?

**Comment by extend1994** [comment]
@alexandru-paduraru yes, it's okay :) You mentioned you're going to remove images, would it be also alright if we don't include images in the old versions?

**Comment by sufuf3** [comment]
@extend1994 I think the author answered in this comment https://github.com/cdnjs/cdnjs/pull/11596#issuecomment-324072108
Your question is the same with https://github.com/cdnjs/cdnjs/pull/11596#issuecomment-324065663
Thank you.

**Comment by extend1994** [comment]
@sufuf3 I know the author answered your question. I just thought both of you are talking about the future releases? So I asked if images files could be also excluded in the old versions because old versions had been released and images were used.

**Comment by alexandru-paduraru** [comment]
@extend1994 @sufuf3 yes, we can remove them from old versions too. Where can I see the final version hosted on cdnjs so I can check if everything is working fine? If there is something missing we will do the update and I think it will be automatically updated in cdnjs too. I'm right?

**Comment by sufuf3** [comment]
@alexandru-paduraru Thank you! 👍

**Comment by extend1994** [comment]
@alexandru-paduraru Thank a lot! After this PR is merged, you can see the files on https://cdnjs.com/libraries/material-kit and the source on GitHub will be http://github.com/cdnjs/cdnjs/blob/master/ajax/libs/material-kit.

File: ajax/libs/material-kit/1.1.0/css/material-kit.css [hunk 4]
```diff
@@ -0,0 +1,3761 @@
...
+  font-weight: 400;
+}
+.form-group.form-group-sm label.control-label {
+  font-size: 9px;
+  line-height: 1.125;
+  color: #AAAAAA;
+  font-weight: 400;
+  margin: 16px 0 0 0;
+}
+.f
```
Comment (PeterDaveHello) [inline ajax/libs/material-kit/1.1.0/css/material-kit.css]: You should add them back as they are referenced here, the future removal is the future thing.

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 I added `img/arrow*` files and manually added old versions because the current auto-update config can't get `img/arrow*`
Please help me review this PR, Thank you.

File: ajax/libs/material-kit/package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/creativetimofficial/material-kit.git",
+    "fileMap": [
+      {
+        "basePath": "assets",
+        "files": [
+          "+(js|css)/material-kit*"
```
Comment (extend1994) [inline ajax/libs/material-kit/package.json]: Using `img/**/*.+(png|cur)` can grab images.
[Pattern ref1](#discussion_r134392541)
[Pattern ref2](#issuecomment-323942289)

**Comment by sufuf3** [comment]
@extend1994 
As this comment https://github.com/cdnjs/cdnjs/pull/11596#discussion_r136946008 I think I just need to add the img files.
The the auto-update config can get the files what the author said https://github.com/cdnjs/cdnjs/pull/11596#issuecomment-326925411 is the future.
As 
> the future removal is the future thing.
said.

What do you think?

**Pull request merged**